server/src/main/java/org/elasticsearch/search/profile/query/ProfileWeight.java

```diff
@@ -47,7 +47,8 @@ public Scorer scorer(LeafReaderContext context) throws IOException {
 
     @Override
     public ScorerSupplier scorerSupplier(LeafReaderContext context) throws IOException {
-        Timer timer = profile.getTimer(QueryTimingType.BUILD_SCORER);
+        // TODO this is called concurrently now but we only have one timer!
+        final Timer timer = profile.getNewTimer(QueryTimingType.BUILD_SCORER);
```

Comment:
is the TODO still valid or is it addressed by your changes?

Reply:
This can go.